ibis/backends/polars/__init__.py

```diff
@@ -0,0 +1,194 @@
+from __future__ import annotations
+
+import importlib
+import re
+from functools import lru_cache
+from pathlib import Path
+from typing import Any, Mapping, MutableMapping
+
+import pandas as pd
+import pyarrow as pa
+
+import ibis.common.exceptions as com
+import ibis.expr.analysis as an
+import ibis.expr.operations as ops
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+from ibis.backends.base import BaseBackend
+from ibis.backends.datafusion.compiler import translate
```

Comment:
I realize from my previous comment it may not be clear what to instead. I think the best option is to copy whatever you can from `ibis/backends/datafusion/compiler.py` and implement the various `translate` rules as needed, in the polars-y way. It might be repetitive, but it removes the datafusion dependency.

Now that we have a second backend in the same spirit as DataFusion we can explore ways to consolidate the rules independent of the specific backend, to avoid repetition.

Reply:
Yes, I have to still understand this a bit. 

I was thinking of building it around a polars `LazyFrame` which is an abstraction over a computation.

It only gets materialized into a `DataFrame` after a `collect` call. I don't know what's best format for this.

But if I am a bit further I will ask more directed questions.